### ISSUE TITLE
Problems with using Depends FastApi method without Provide in wired module

### ISSUE THREAD

**Issue by Lojka-oops** [ets-labs/python-dependency-injector#330]
Hello, I am having difficulty implementing authorization using FastApi and Dependency Injection Framework. I am trying to do something like this:

```
from fastapi import Depends, APIRouter
from fastapi.security import HTTPBasic, HTTPBasicCredentials
from dependency_injector.wiring import inject, Provide

from services.some_service import SomeService
from containers import Container

router = APIRouter()
security = HTTPBasic()


@router.get("/users/me")
@inject
def read_current_user(
    some_service: SomeService = Depends(Provide[Container.some_service]),
    credentials: HTTPBasicCredentials = Depends(security)
):
    return {"username": credentials.username, "password": credentials.password}
```

But I am getting an error: `AttributeError: 'HTTPBasic' object has no attribute 'provider'`. I tried to add `HTTPBasic` to the container and use from there, it fixed the error, but the authorization still didn't work. I am clearly missing something. Is there a template for this? Thanks in advance for your help with this issue!

**Comment by rmk135** [comment]
Hey @Lojka-oops. Not sure what causes the problem. I'll debug it later today.

**Comment by Lojka-oops** [comment]
Thank you!

**Comment by rmk135** [comment]
Hey @Lojka-oops , confirming that's a bug. Fixing

**Comment by rmk135** [comment]
I have released version `4.5.2` with a fix. Please, upgrade. Thank you for reporting the issue.

**Comment by Lojka-oops** [comment]
Thank you very much for your help!

**Comment by juanmarin96** [comment]
I saw that the merge release 4.5.2 failed... When do you believe that it can be released again?, P.D: Thanks a lot for fixed it!!

**Comment by juanmarin96** [comment]
Hi, @Lojka-oops can you show me how are you working with dependency injector in you FastAPI project, I am using the 4.4.1 version because if I update it my project get broken and show: "AttributeError: 'Provide' object has no attribute 'get_multi'":

![image](https://user-images.githubusercontent.com/19893284/101260174-f597f580-36fb-11eb-96f3-19306835e0a5.png)

**Comment by Lojka-oops** [comment]
Hi, @juanmarin96, It's how I working with dependency injector:

containers.py
```
from dependency_injector import containers, providers

from services.app_service import AppService


class Container(containers.DeclarativeContainer):

    config = providers.Configuration()

    app_service = providers.Factory(
        AppService
    )
```
app_controller.py
```
from dependency_injector.wiring import inject, Provide
from fastapi import APIRouter, Depends,

from schemas import apps_schemas
from services.app_service import AppService
from containers import Container

router = APIRouter(tags=['apps'])


@router.post("/apps", response_model=apps_schemas.AppBaseSchema, status_code=201)
@inject
async def create_app(
    app: apps_schemas.AppCreateSchema, 
    app_service: AppService = Depends(Provide[Container.app_service])
):
    app = await app_service.create_app(app)
    
    return app
```

I hope this can help you.

**Comment by Lojka-oops** [comment]
@juanmarin96, my mistake I also got this error after update

**Comment by juanmarin96** [comment]
@Lojka-oops yeah, it's a bug I already report it, thanks for you answer.

**Comment by rmk135** [comment]
@Lojka-oops, Please, upgrade to ``4.5.3``. It fixes `4.5.2` degradation. I've also realised that there were no FastAPI tests and added them. It will help to prevent such issues.

My apologies for `4.5.2` release.

**Comment by Lojka-oops** [comment]
Thanks again for your help!
